server/server.go

```diff
@@ -1,46 +1,32 @@
 package server
 
 import (
-	"database/sql"
 	"html/template"
 	"log"
 	"net/http"
 
+	"github.com/OAuth2withJWT/identity-provider/app"
 	"github.com/gorilla/mux"
-	"golang.org/x/crypto/bcrypt"
 )
 
 type Server struct {
-	r  *mux.Router
-	db *sql.DB
+	router      *mux.Router
+	userService *app.UserService
```

Comment:
Why not keep a pointer to the application instance? The point of a single struct (`app.Application`) is to keep references to all the services in a single struct, so the server can just have `app.Application` as a dependency and it's easier to inject a single struct. For example:
```go
type Server struct {
    // ...
    app *app.Application
}
```